doc/userguide/rules/intro.rst

```diff
@@ -30,172 +32,197 @@ In this example, :example-rule-action:`red` is the action,
+Suricata instance.
+
 Action
 ------
-
-For more information read 'Action Order' see
-:ref:`suricata-yaml-action-order`.
-
 .. container:: example-rule
 
     :example-rule-emphasis:`drop` tcp $HOME_NET any -> $EXTERNAL_NET any (msg:"ET TROJAN Likely Bot Nick in IRC (USA +..)"; flow:established,to_server; flowbits:isset,is_proto_irc; content:"NICK "; pcre:"/NICK .*USA.*[0-9]{3,}/i"; reference:url,doc.emergingthreats.net/2008124; classtype:trojan-activity; sid:2008124; rev:2;)
 
-Protocol
---------
-
-This keyword in a signature tells Suricata which protocol it
-concerns. You can choose between four settings.  tcp (for
-tcp-traffic), udp, icmp and ip. ip stands for 'all' or 'any'.
-Suricata adds a few protocols : http, ftp, tls (this includes ssl),
-smb and dns (from v2.0). These are the so-called application layer
-protocols or layer 7 protocols.  If you have a signature with for
-instance a http-protocol, Suricata makes sure the signature can only
-match if it concerns http-traffic.
+For more information see :ref:`suricata-yaml-action-order`.
 
-Example:
 
+Protocol
+--------
 .. container:: example-rule
 
     drop :example-rule-emphasis:`tcp` $HOME_NET any -> $EXTERNAL_NET any (msg:"ET TROJAN Likely Bot Nick in IRC (USA +..)"; flow:established,to_server; flowbits:isset,is_proto_irc; content:"NICK "; pcre:"/NICK .*USA.*[0-9]{3,}/i"; reference:url,doc.emergingthreats.net/2008124; classtype:trojan-activity; sid:2008124; rev:2;)
 
-In this example the red, bold-faced part is the protocol.
-
-Source and destination
-----------------------
-
-In source you can assign IP-addresses; IPv4 and IPv6 combined as well
-as separated. You can also set variables such as HOME_NET. (For more
-information see :ref:`suricata-yaml-rule-vars`. In the Yaml-file you
-can set IP-addresses for variables such as EXTERNAL_NET and
-HOME_NET. These settings will be used when you use these variables in
-a rule.  In source and destination you can make use of signs like !
-And [ ].
-
-For example::
-
-  ! 1.1.1.1                       (Every IP address but 1.1.1.1)
-  ![1.1.1.1, 1.1.1.2]             (Every IP address but 1.1.1.1 and 1.1.1.2)
-  $HOME_NET                       (Your setting of HOME_NET in yaml)
-  [$EXTERNAL_NET, !$HOME_NET]     (EXTERNAL_NET and not HOME_NET)
-  [10.0.0.0/24, !10.0.0.5]        (10.0.0.0/24 except for 10.0.0.5)
-  […..,[....]]
-  […. ,![.....]]
-
-
-Pay attention to the following:
+This keyword in a signature tells Suricata which protocol it
+concerns. You can choose between four basic protocols:
 
-If your settings in Yaml are::
+* tcp (for tcp-traffic)
+* udp
+* icmp
+* ip (ip stands for 'all' or 'any')
 
-  HOME_NET: any
-  EXTERNAL_NET: ! $HOME_NET
+There are also a few so-called application layer protocols, or layer 7 protocols
+you can pick from. These are:
 
-You can not write a signature using EXTERNAL_NET because it stands for
-'not any'. This is a invalid setting.
+* http
+* ftp
+* tls (this includes ssl)
+* smb
+* dns (from v2.0)
```

Comment:
we also have smtp for example, icmp and ssh are also in ET rulesets.

Reply:
Are these keywords supported for the protocol field as well? As there's no list anywhere.

Reply:
Yes, those are at the protocol field. We have some more, see also https://suricata-ids.org/features/all-features/ but adding that in a seperate commit/PR is totally fine with me.